ci/build-and-test-nupic-bindings.sh

```diff
@@ -0,0 +1,171 @@
+
+  test results: nupic.bindings test results will be located in the subdirectory
+                test_results of the source tree's root directory with the
+                the following content:
+
+                cplusplus-test-results.txt
+                junit-test-results.xml
+                htmlcov-report/
+
+"
+
+if [[ $1 == --help ]]; then
+  echo "${USAGE}"
+  exit 0
+fi
+
+if [[ $# > 0 ]]; then
+  echo "ERROR Unexpected arguments: ${@}" >&2
+  echo "${USAGE}" >&2
+  exit 1
+fi
+
+
+set -o xtrace
+
+
+# Apply defaults
+BUILD_TYPE=${BUILD_TYPE-"Release"}
+
+
+NUPIC_CORE_ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )"
+
+DEST_WHEELHOUSE="${NUPIC_CORE_ROOT}/nupic_bindings_wheelhouse"
+
+TEST_RESULTS_DIR="${NUPIC_CORE_ROOT}/test_results"
+
+echo "RUNNING NUPIC BINDINGS BUILD: BUILD_TYPE=${BUILD_TYPE}, " \
+     "RESULT_KEY=${RESULT_KEY}, DEST_WHEELHOUSE=${DEST_WHEELHOUSE}" >&2
+
+# Install pycapnp to get the matching capnproto headers for nupic.core build
+# NOTE Conditional pycapnp dependency should be incorporated into
+# bindings/py/requirements.txt to abstract it from upstream scripts.
+pip install pycapnp==0.5.8
+
+# Install nupic.bindings dependencies; the nupic.core cmake build depends on
+# some of them (e.g., numpy).
+pip install -r ${NUPIC_CORE_ROOT}/bindings/py/requirements.txt
+
+
+#
+# Build nupic.bindings
+#
+
+# ZZZ debug statement to see if build/scripts was inherited from another job
+ls ${NUPIC_CORE_ROOT}/build/scripts || true
+# ZZZ end debug
+
+mkdir -p ${NUPIC_CORE_ROOT}/build/scripts
+cd ${NUPIC_CORE_ROOT}/build/scripts
+
+# Configure nupic.core build
+if [[ "$BUILD_TYPE" == "Debug" ]]; then
+  EXTRA_CMAKE_DEFINITIONS="-DNUPIC_IWYU=ON -DNTA_COV_ENABLED=ON"
+fi
+
+cmake ${NUPIC_CORE_ROOT} \
+    -DCMAKE_BUILD_TYPE=${BUILD_TYPE} \
+    ${EXTRA_CMAKE_DEFINITIONS} \
+    -DCMAKE_INSTALL_PREFIX=${NUPIC_CORE_ROOT}/build/release \
+    -DPY_EXTENSIONS_DIR=${NUPIC_CORE_ROOT}/bindings/py/nupic/bindings
+
+# Build nupic.core
+make install
+
+# Build nupic.bindings python extensions from nupic.core build artifacts
+if [[ $WHEEL_PLAT ]]; then
+  EXTRA_WHEEL_OPTIONS="--plat-name ${WHEEL_PLAT}"
+fi
+
+cd ${NUPIC_CORE_ROOT}
+python setup.py bdist_wheel --dist-dir ${DEST_WHEELHOUSE} ${EXTRA_WHEEL_OPTIONS}
+
+
+#
+# Test
+#
+
+# ZZZ debug statement to see if test_results was inherited from another job
+ls -R ${TEST_RESULTS_DIR} || true
+# ZZZ end debug
+
+mkdir ${TEST_RESULTS_DIR}
+
+# Install the wheel that we just built
+pip install --ignore-installed ${DEST_WHEELHOUSE}/nupic.bindings-*.whl
+
+# Run the nupic.core C++ tests
+cd ${NUPIC_CORE_ROOT}/build/release/bin
+(
+  ./connections_performance_test
+  ./cpp_region_test
+  ./helloregion
+  ./hello_sp_tp
+  ./prototest
+  ./py_region_test
+  ./unit_tests
+) 2>&1 | tee "${TEST_RESULTS_DIR}/cplusplus-test-results.txt"
+
+
+py.test --verbose \
+  --junitxml \
+    "${TEST_RESULTS_DIR}/junit-test-results.xml" \
+  --cov nupic.bindings \
```

Comment:
It might be cleaner to create a .coveragerc file that captures these things. You can also specify the html output location [1] in that file so you don't have to do the `mv` afterwards. And it creates an explicit convention around how to generate and where to save the coverage results.

[1] http://stackoverflow.com/a/27537412